tensorflow_addons/losses/tests/lifted_test.py

```diff
@@ -100,6 +108,44 @@ def test_lifted_struct():
     np.testing.assert_almost_equal(loss.numpy(), loss_np, 3)
 
 
+def test_dtypes_float16():
+    num_data = 10
+    feat_dim = 6
+    margin = 1.0
+    num_classes = 4
+
+    embedding = np.random.rand(num_data, feat_dim).astype(np.float16)
+    labels = np.random.randint(0, num_classes, size=num_data).astype(np.float16)
+
+    loss_np = lifted_struct_loss_np(labels, embedding, margin)
+
+    # Compute the loss in TF.
+    y_true = tf.constant(labels)
+    y_pred = tf.constant(embedding, dtype=tf.float16)
+    cce_obj = lifted.LiftedStructLoss()
+    loss = cce_obj(y_true, y_pred)
+    np.testing.assert_allclose(loss, loss_np, rtol=1e-3, atol=1e-3)
+
+
+def test_dtypes_bfloat16():
+    num_data = 20
+    feat_dim = 6
+    margin = 1.0
+    num_classes = 4
+
+    embedding = np.random.rand(num_data, feat_dim).astype(np.float16)
+    labels = np.random.randint(0, num_classes, size=num_data).astype(np.float16)
+
+    loss_np = lifted_struct_loss_np(labels, embedding, margin)
+
+    # Compute the loss in TF.
+    y_true = tf.constant(labels)
+    y_pred = tf.constant(embedding, dtype=tf.bfloat16)
+    cce_obj = lifted.LiftedStructLoss()
+    loss = cce_obj(y_true, y_pred)
+    np.testing.assert_allclose(loss.numpy(), loss_np, rtol=1e-2, atol=1e-2)
```

Comment:
How about using `@pytest.mark.parametrize` to and `assert_allclose_according_to_type` to test against different precisions? It makes less duplicated codes :-)

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/tests/kappa_loss_test.py#L70
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/test_utils.py#L221

Reply:
Yes, makes it a lot cleaner.

`assert_allclose_according_to_type` does not check for bfloat16. Which rtol and atol is best used for bfloat16? The triplet loss test with bfloat16 fails for rtol=1e-3 and atol=1e-3 and accepts for rtol=1e-2 and atol=1e-2.

Reply:
Hi, I modify `assert_allclose_according_to_type` to make it support bfloat16. See #1698. You could go ahead after it gets merged. Thank you :-)

Reply:
Ok, thank you :)